src/librustc_data_structures/sync.rs

```diff
@@ -130,7 +131,19 @@ cfg_if! {
         #[macro_export]
         macro_rules! parallel {
             ($($blocks:tt),*) => {
-                $($blocks)*;
+                let mut panic = None;
+                $(
+                    if let Err(p) = ::std::panic::catch_unwind(
+                        ::std::panic::AssertUnwindSafe(|| $blocks)
+                    ) {
+                        if panic.is_none() {
+                            panic = Some(p);
+                        }
+                    }
+                )*
+                if let Some(panic) = panic {
+                    ::std::panic::resume_unwind(panic);
+                }
```

Comment:
What's going on here? Is this to make behaviour more consistent with parallel execution?

Reply:
Yes. Otherwise the single threaded compiler will stop at the first fatal error and the test output will diverge from the parallel compiler.